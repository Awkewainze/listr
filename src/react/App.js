import React from 'react';
import './App.scss';
import Header from './common/Header';
import Footer from './common/Footer';
import { Switch, Route } from 'react-router-dom';
import ActiveListContainer from './list/ActiveListContainer';

function App() {
  return (
    <div className="App">
      <Header/>
      <Switch>
        <Route path="/" component={ActiveListContainer} />
      </Switch>
      <Footer/>
    </div>
  );
}

export default App;
