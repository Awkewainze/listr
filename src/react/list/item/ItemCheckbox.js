import React from 'react';
import Checkbox from '../../common/Checkbox';

const ItemCheckbox = ({
  className,
  checked,
  index,
  itemId,
  onChange
}) => {
  const onToggle = event => {
    event.preventDefault();
    onChange(itemId, event);
  };
  const id = `Checkbox-${index + 1}`;

  return (
    <Checkbox
      className={className}
      onChange={onToggle}
      id={id}
      checked={checked}
    />
  );
};

export default ItemCheckbox;