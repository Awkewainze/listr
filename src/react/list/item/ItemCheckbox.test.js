import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import ItemCheckbox from './ItemCheckbox';

const testItemId = '123123';

const TEST_PROPS = {
  onChange: jest.fn(),
  index: 1,
  itemId: testItemId,
  checked: false
};

it('matches snapshot', () => {
  const { container } = render(<ItemCheckbox { ...TEST_PROPS } />);
  expect(container.firstChild).toMatchSnapshot();
});

it('calls onChange when checked or unchecked', () => {
  const props = {
    ...TEST_PROPS
  };
  const { container } = render(<ItemCheckbox { ...props } />);
  fireEvent.click(container.querySelector('label'));
  const { onChange } = TEST_PROPS;

  expect(onChange).toHaveBeenCalledTimes(1);
});