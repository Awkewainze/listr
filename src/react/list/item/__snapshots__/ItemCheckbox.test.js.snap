// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<label
  class="Checkbox  "
>
  <input
    class="Checkbox__hidden"
    id="Checkbox-2"
    type="checkbox"
  />
  <svg
    class="Checkbox__icon"
    viewBox="0 0 24 24"
  >
    <polyline
      points="20 6 9 17 4 12"
    />
  </svg>
</label>
`;
